framework/doc/content/source/meshgenerators/TransfiniteMeshGenerator.md

```diff
@@ -0,0 +1,70 @@
+illustrates the use of the `TransfiniteMeshGenerator` object to construct general quadrilaterals as in Fig. 2. By default the distribution of points along edges is considered equidistant, unless specified otherwise by the user using the parameters `bias_x` or `bias_y`. The core of the algorithm interpolates/fills up the inner domain for any edge type, curved, parametrized, discrete. The current implementation considers all these cases with special attention given to arcs of a circle which are widely encountered in scientific computing applications. 
+
+The convention for the corners numbering should follow Fig. 1, prescribed in the order +(0,0) - (0,1) - (1,1) - (1,0)+. The direction along edges, important if the user provides them using the DISCRETE option, should follow the direction of the arrows as in Fig. 1.
+
+
+!media large_media/framework/meshgenerators/transfinite_quadrilateral_sm.png style=float:right;margin:2%;width:20%; caption=Fig. 2: A quadrilateral with one edge on an arc circle.
+
+
+This generator allows any type of curvilinear edges. In Fig. 2 we show a mesh where the bottom edge has an inward arc circle. However, this algorithm generates a mesh for any parametrized edge, or given set of points on an edge. For arcs of a circle, since they are widely encountered in applications, we customized an approach that requires a user to specify only the deviation from a straight line in the middle of an edge. 
+
+The default distribution of points on an edge is equidistant and nothing expplicit needs to be specified. However, for meshes that need to capture more scales in different regions we have two additional options that can be prescribed.  The distribution of points on an edge can be either provided as a discrete set of points, using the edge type `DISCRETE`, or as a bias along the edge. The only requirement is that opposite edges have the same number of points.
+
+## Example 
+
+
+!media large_media/framework/meshgenerators/transfinite_arccircle_annular_sm.png style=float:right;margin:2%;width:18%; caption=Fig. 3: Annulus section generated using `CIRCARC` edges.
+
+
+&nbsp;
+
+This mesh generator considers 4 types of edges defined in `edge_type`, where `edge` is either left, right, top or bottom.
+
+  - `LINE`, constructs a line between provided corner vertices. A quadrilater can be obtained only with line sides as in Fig. 2.
+
+  - `CIRCARC`, constructs an arc of a circle and requires an additional parameter, named `edge_parameter`. This parameter can be the distance at the middle of an edge between a straight line and the arc of circle desired, as in the test [arccircle_generator.i](test/tests/meshgenerators/transfinite_generator/arcircle_generator.i). The edge is then generated by computing the middle point on the edge. The deviation from a straight line is set by convention to be positive if the arc circle is outward, or negative if inward, in this fashion the user can specify different orientations, as in Fig. 4.
```

Comment:
Could you add a couple of lines to show the example calculations in LaTeX for both the distance and midpoint options (as you did to help me via slack)? I'd recommend something as simple as:
in 2D for the distance option:
$$
edge_parameter = 'radius * (1.0 - 1.0 / sqrt(2.0))'
$$

and in 2D for the midpoint option:
$$
edge_parameter = 'origin_x + radius * cos(\theta) && origin_y + radius * sin(\theta) && 0.0'
$$

Reply:
They're not supposed to be, let me check.